libs/utils/decorate.js

```diff
@@ -331,20 +410,84 @@ export async function loadCDT(el, classList) {
   }
 }
 
+export function isAccessible(anchorTag) {
+  return !anchorTag.hash.includes(HIDE_CONTROLS);
+}
+
+function updateFirstVideo() {
+  if (firstVideo != null && firstVideo?.controls === false && videoCounter > 1) {
+    let videoHolder = document.querySelector('[video-index="1"]') || firstVideo.closest('.video-holder');
+    if (videoHolder.nodeName !== 'A') {
+      videoHolder = videoHolder.querySelector('a.pause-play-wrapper');
+    }
+    const firstVideoLabel = videoHolder.getAttribute('aria-label');
+    videoHolder.setAttribute('aria-label', `${firstVideoLabel} 1`);
+    firstVideo = null;
+  }
+}
+
+function updateAriaLabel(videoEl, videoAttrs) {
+  if (!videoEl.getAttributeNames().includes('data-hoverplay')) {
+    const pausePlayWrapper = videoEl.parentElement.querySelector('.pause-play-wrapper') || videoEl.closest('.pause-play-wrapper');
+    const pauseIcon = pausePlayWrapper.querySelector('.pause-icon');
+    const playIcon = pausePlayWrapper.querySelector('.play-icon');
+    const indexOfVideo = pausePlayWrapper.getAttribute('video-index');
+    const ariaLabel = `${videoAttrs.includes('autoplay') ? videoLabels.pauseMotion : videoLabels.playMotion} ${indexOfVideo === '1' && videoCounter === 1 ? '' : indexOfVideo}`.trim();
+    pausePlayWrapper.setAttribute('aria-label', ariaLabel);
+    pauseIcon.setAttribute('alt', videoLabels.pauseMotion);
+    playIcon.setAttribute('alt', videoLabels.playMotion);
+    updateFirstVideo();
+  }
+}
+
+export function decoratePausePlayWrapper(videoEl, videoAttrs) {
+  if (!videoLabels.hasFetched) {
+    Promise.all([import('../features/placeholders.js'), import('../blocks/global-navigation/utilities/utilities.js')]).then(([{ replaceKeyArray }, { getFedsPlaceholderConfig }]) => {
```

Comment:
Why do you import `getFedsPlaceholderConfig` from the global-navigation? Should that move into `federated.js` ?

A dependency from `decorate` onto the gnav utilities sounds wrong